quill-core/src/main/scala/io/getquill/quotation/Parsing.scala

```diff
@@ -272,6 +273,15 @@ trait Parsing {
       case "!=" => EqualityOperator.`!=`
     }
 
+  val optionSingleOrderOperationParser: Parser[Operation] = Parser[Operation] {
+    case q"$a.nonEmpty" if is[Option[Any]](a) =>
+      BinaryOperation(astParser(a), EqualityOperator.`!=`, NullValue)
+    case q"$a.isDefined" if is[Option[Any]](a) =>
+      BinaryOperation(astParser(a), EqualityOperator.`!=`, NullValue)
+    case q"$a.isEmpty" if is[Option[Any]](a) =>
+      BinaryOperation(astParser(a), EqualityOperator.`==`, NullValue)
```

Comment:
The transformation to `null` comparison isn't a generic operation and should be done by the submodules. We could support target languages that have proper optional value handling, which wouldn't use null comparisons. 

The operations must produce new AST types and be translated to null comparisons by the sql module.